### PR TITLE
adding correct area attributes to proportions_lc_by_prms() functions

### PR DESCRIPTION
**Pull request by msleckman**
Added 2 params - area_col and length_col - in the `proportions_lc_by_prms()` saved in `combine_NLCD_PRMS.R`. 
```
proportion_lc_by_prms <- function(NLCD_LC_df_w_area, area_col, length_col = 'LENGTHKM', catchment_att = 'CAT')
```

This way when aggregating to PRMS scale, the NLCD TOT datasets is subsetting and aggregating using the right area metric comparedi to NLCD ACC and NLCD CAT. 

See one in-line comment regarding ACC.

Addresses issue #130

### CODE REVIEW

File: 2_process.R [hunk 1]
```diff
@@ -129,6 +133,8 @@ p2_targets_list <- list(
     proportion_lc_by_prms(p2_NLCD_LC_w_catchment_area %>%
                             # filtering to only the comid_downs of each PRMS - nrow = ~459
                             filter(comid %in% p2_drb_comids_down$comid),
+                          area_col = 'TOTDASQKM',
```
Comment (msleckman) [inline 2_process.R:136]: for NLCD ACC variables, do we want to take the area of the catchment (AREASQKM) col or the Total upstream area (TOTDAKM) ?

File: 2_process/src/combine_NLCD_PRMS.R [hunk 2]
```diff
@@ -27,15 +27,17 @@ AOI_LC_w_area <- function(NLCD_LC_df, area_att, aoi_comids_df){
...
 #' @param NLCD_LC_df_w_area df output of AOI_LC_w_area that has LC data per COMID & PRMS id and area in (km2)
+#' @param area_col str col with catchment ('AREASQKM') or total upstream catchment area ('TOTDASQKM')
+#' @param length_col str name of col with length of catchment area ('LENGTHKM). Default 'LENGTHKM'
 #' @param @param catchment_att Specifies the catchment attribute to subset to. Must be either 'CAT', "ACC", 'TOT'. Default = "CAT"
-#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_LC_w_catchment_area)
+#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_NLCD_LC_w_catchment_area, area_col = 'AREASQKM')
 
   area_df <- NLCD_LC_df_w_area %>%
-    mutate(AREASQKM_approx = case_when(AREASQKM == 0 ~ LENGTHKM^2, TRUE  ~ AREASQKM)) %>% 
+    mutate(AREASQKM_approx = case_when(.data[[area_col]] == 0 ~ .data[[length_col]]^2, TRUE  ~ .data[[area_col]])) %>%
```
Comment (jds485) [inline 2_process/src/combine_NLCD_PRMS.R:40]: This approximation will not work for TOTDASQKM. I think we should only use an approximate area if area_col == AREASQKM

Did you check if there are any TOTDASQKM = 0?

Comment (msleckman) [reply 2_process/src/combine_NLCD_PRMS.R:40]: Yes, based on the source targets `p2_NLCD_LC_w_catchment_area`: 
42 comids have 0 TOTAL AREA. That amounts to 36 unique PRMS_segids. 

```
> p2_NLCD_LC_w_catchment_area %>% filter(TOTDASQKM == 0) %>% select(PRMS_segid, comid, AREASQKM, TOTDASQKM)
# A tibble: 42 × 4
   PRMS_segid comid   AREASQKM TOTDASQKM
   <chr>      <chr>      <dbl>     <dbl>
 1 1264_1     8077104        0         0
 2 1264_1     8077858        0         0
 3 1265_1     8078052        0         0
 4 141_1      2617690        0         0
 5 164_1      2743484        0         0
 6 17_1       1750665        0         0
 7 2129_1     9482534        0         0
 8 222_1      4152664        0         0
 9 238_1      4153808        0         0
10 240_1      4154548        0         0
# … with 32 more rows

```

**Comment by msleckman** [comment]
Waiting on #122 for merge.

**Comment by msleckman** [comment]
@jds485 cannot say whether this needs to be merged now. Can you review based on most recent changes to the pipeline?

**Comment by msleckman** [comment]
> @jds485 cannot say whether this needs to be merged now. Can you review based on most recent changes to the pipeline?

Nevermind this is also ready to merge.

**Review by jds485 — APPROVED** [review]
Looks good! You can merge when you're ready

**Comment by msleckman** [comment]
Clarification point: 

NLCD proportions for total and accumulated are only aggregated from its respective `comid down`: 
in `2_process.R`, added a filter to only consider these down catchments:

```
proportion_lc_by_prms(**p2_NLCD_LC_w_catchment_area %>%**
                            # filtering to only the comid_downs of each PRMS - nrow = ~459
                            **filter(comid %in% p2_drb_comids_down$comid)**,
                          catchment_att = 'ACC', remove_NODATA_cols = TRUE)
                          area_col = 'TOTDASQKM',length_col = 'LENGTHKM',
                          catchment_att = 'ACC', 
                          remove_NODATA_cols = TRUE)
```

**Pull request merged**